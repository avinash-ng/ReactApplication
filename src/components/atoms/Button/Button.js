import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
