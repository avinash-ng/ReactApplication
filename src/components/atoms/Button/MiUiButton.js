import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton