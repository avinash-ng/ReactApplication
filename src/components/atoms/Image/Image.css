img {
    width: 450px;
    height: 450px;
    float: left;
    margin-left: 20px;
    margin-top: 0px;
    padding-top: 60px;
    /* margin-bottom: 30px; */
}