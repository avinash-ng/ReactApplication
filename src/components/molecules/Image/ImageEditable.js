import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable