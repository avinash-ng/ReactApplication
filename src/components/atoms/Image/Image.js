import React from 'react'
import PropTypes from "prop-types"
import "./Image.css"
const Image = (props) => {
    return (
        <img data-testid="image" src={props.imageName} alt={props.altText} />
    )
}
Image.propTypes = {
    imageName: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
} 

export default Image