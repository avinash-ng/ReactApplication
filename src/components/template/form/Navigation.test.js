const { shallow, configure, mount } = require("enzyme")
import Navigation from "./Navigation"
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import Image from "../../atoms/Image/Image";
import ImageEditable from "../../molecules/Image/ImageEditable";
import { ListItem, AppBar } from '@material-ui/core'
import { TabPanel }  from "./Navigation";

configure({ adapter: new Adapter() });
describe('testing <Navigation /> ', () => {
    it('renders <AppBar /> correctly', () => {

        const RenderComponent = shallow(<Navigation > </Navigation>)
        expect(RenderComponent.find(AppBar).exists()).toBe(true)
    })

    it('renders <TabPanel /> correctly', () => {
        const RenderComponent = shallow(<Navigation > </Navigation>)
        expect(RenderComponent.find(TabPanel).exists()).toBe(true)
    })

   

    it('matches snapshot', () => {
        const RenderComponent = shallow(<Navigation ></Navigation>)
        expect(RenderComponent).toMatchSnapshot()
    })
})