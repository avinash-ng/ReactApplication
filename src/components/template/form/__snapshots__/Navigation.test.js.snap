// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Navigation />  matches snapshot 1`] = `ShallowWrapper {}`;
